client/quantum_serverless/quantum_serverless.py

```diff
@@ -85,12 +85,12 @@ def job_client(self):
         """Job client for given provider."""
         return self._selected_provider.job_client()
 
-    def run_program(self, program: Program) -> Optional[Job]:
-        """Execute program as a async job
+    def run_nested_program(self, nested_program: NestedProgram) -> Optional[Job]:
```

Comment:
The idea @agarwal-aryan is change this line to:

```suggestion
    def run(self, nested_program: NestedProgram) -> Optional[Job]:
```

And review where is being used.